test/files/pos/t6317.scala

```diff
@@ -0,0 +1,15 @@
+abstract class C { 
+  def overloaded(foo: String, bar: String): String
+  def overloaded(foo: String, bar: String, baz: String): Unit
+}
+
+class ScalaCompilerKiller {
+  implicit def CWrapper(c: C) = new {
+    def overloaded(request: Any): Unit = {}
+  }
+
+  val sps = List[(String, String)]()
+
+  // to repro, need: implicit conversion, overloading, pair in synthetic scrutinee in function passed to higher-order method
+  (null: C).overloaded(sps.map(_ match { case (r, _) => r }))
```

Comment:
If I understand [the comment correctly](https://github.com/scala/bug/issues/6317#issuecomment-292411079), the bug was triggered when `_ match` was missing, and the workaround was to make it explicit. Let's just include both versions in the test.

Reply:
Yeah, making the workaround the test case wasn't very bright.